src/main/java/com/google/codeu/servlets/MessageServlet.java

```diff
@@ -102,11 +105,23 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     String textWithImagesReplaced = userText.replaceAll(regex, replacement);
 
     Parser parser = Parser.builder().build();
-    Node document = parser.parse(textWithImagesReplaced);
     HtmlRenderer renderer = HtmlRenderer.builder().build();
-    String sanitizedText = renderer.render(document);
 
+    Node document = parser.parse(textWithImagesReplaced);
+    String sanitizedText = renderer.render(document);
     Message message = new Message(user, sanitizedText, recipient, sentimentScore);
+
+    Pattern emailRegex =
+        Pattern.compile(
+            "@(?:[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*|\"(?:[\\x01-\\x08\\x0b\\x0c\\x0e-\\x1f\\x21\\x23-\\x5b\\x5d-\\x7f]|\\\\[\\x01-\\x09\\x0b\\x0c\\x0e-\\x7f])*\")@(?:(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?|\\[(?:(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\\.){3}(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?|[a-z0-9-]*[a-z0-9]:(?:[\\x01-\\x08\\x0b\\x0c\\x0e-\\x1f\\x21-\\x5a\\x53-\\x7f]|\\\\[\\x01-\\x09\\x0b\\x0c\\x0e-\\x7f])+)\\])");
```

Comment:
Might be unhandled by the formatter? Should I break into <80 char lines and concatenate?